src/main/java/com/google/sps/servlets/EventServlet.java

```diff
@@ -215,13 +230,29 @@ private void createEntity(
 
     // Get formatted dates and times for display.
     String dateTimeFormatted = createDateTimeDisplay(startDate, startTimeFormatted, 
-          endDate, endTimeFormatted, timeZone);
+          endDate, endTimeFormatted);
 
     Entity eventEntity = new Entity(Constants.EVENT_ENTITY_PARAM);
+    if (isEventEditing.equals("yes")) {
+      long eventId = Long.parseLong(request.getParameter(Constants.EVENT_ID_PARAM));
+      Query query = new Query(Constants.EVENT_ENTITY_PARAM);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      PreparedQuery results = datastore.prepare(query);
+
+      for (Entity entity : results.asIterable()) {
```

Comment:
It's expensive to iterate over every record in the datastore. Instead consider adding a filter and fetching only the record you're interested in.

https://cloud.google.com/appengine/docs/standard/java/datastore/retrieving-query-results#retrieving_entities